improver_tests/lapse_rate/test_LapseRate.py

```diff
@@ -108,6 +108,25 @@ def test_returns_expected_values(self):
             self.orography)
         self.assertArrayAlmostEqual(result, expected_out)
 
+    def test_returns_expected_values_generator(self):
```

Comment:
There is a lot of duplication between this added test and the other one below.
Suggest reducing duplication between this test method and the one below. The main options I can think are:
- put the bulk of the code in a function, then call that function from these `test_` functions with appropriate parameters
- pytest parametrize
- unittest class inheritance

Reply:
I agree, I haven't done that at the moment, because I want to make sure we're keeping both tests, i.e. if _create_heightdiff_mask is removed, so will its test be.

Reply:
Actually looking at it, I can group the generate tests together (that could be what you meant anyway)